packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -1056,6 +1056,27 @@ export const formNestedStack = async (
   let categories = Object.keys(amplifyMeta);
 
   categories = categories.filter((category) => category !== 'providers');
+
+  const s3 = await S3.getInstance(context);
```

Comment:
can this block of code be extracted into another method? I know this is part of an already massive method, but is there a way to cleanly encapsulate this new logic?

Reply:
That is a good idea. I will do that.